samples/nrf9160/lwm2m_client/Kconfig

```diff
@@ -7,35 +7,143 @@
+
+menu "LwM2M objects"
+menu "Push buttons and on/off switches"
+config APP_PUSH_BUTTON
+	bool "Enable button(s)"
+	select UI_INPUT
+	select LWM2M_IPSO_PUSH_BUTTON
+	select LWM2M_APP_PUSH_BUTTON
+
+
+config APP_ONOFF_SWITCH
+	bool "Enable switches"
+	depends on BOARD_NRF9160DK_NRF9160NS
+	select UI_INPUT
+	select LWM2M_IPSO_ONOFF_SWITCH
+	select LWM2M_APP_ONOFF_SWITCH
+endmenu # Buttons and switches
+
+
+menu "LEDs"
+config APP_LIGHT_CONTROL
+	bool "Enable LED(s)"
+	select UI_LED
+	select LWM2M_IPSO_LIGHT_CONTROL
+	select LWM2M_APP_LIGHT_CONTROL
+
+if APP_LIGHT_CONTROL
+	rsource "src/ui/Kconfig.ui_led"
+endif # APP_LIGHT_CONTROL
+endmenu # LEDs
+
+
+menu "Buzzer"
+config APP_BUZZER
+	bool "Enable buzzer"
+	depends on BOARD_THINGY91_NRF9160NS
+	select UI_BUZZER
+	select LWM2M_IPSO_APP_BUZZER
+	select LWM2M_APP_BUZZER
+
+if APP_BUZZER
+	rsource "src/ipso/Kconfig.ipso_buzzer"
+endif # APP_BUZZER
+endmenu # Buzzer
+
+
+menu "Light sensor"
+config APP_LIGHT_SENSOR
+	bool "Enable light sensor"
+	select LIGHT_SENSOR
+	select LWM2M_IPSO_APP_COLOUR_SENSOR
+	select LWM2M_APP_LIGHT_SENSOR
+
+if APP_LIGHT_SENSOR
+	rsource "src/sensors/Kconfig.light_sensor"
+	rsource "src/ipso/Kconfig.ipso_colour_sensor"
+endif # APP_LIGHT_SENSOR
+endmenu # Light sensor
+
+
+menu "Environment sensors"
+config APP_ENV_SENSOR
+	bool "Enable environment sensors"
+	select ENVIRONMENT_SENSOR
+	help
+		Enables options for enabling temperature,
+		pressure, humidity and gas resistance sensors.
+
+if APP_ENV_SENSOR
+rsource "src/sensors/Kconfig.env_sensor"
+
+config APP_TEMP_SENSOR
+	bool "Enable temperature sensor"
+	select LWM2M_IPSO_TEMP_SENSOR
+	select LWM2M_APP_TEMP_SENSOR
+
+config APP_PRESS_SENSOR
+	bool "Enable air pressure sensor"
+	select LWM2M_IPSO_PRESSURE_SENSOR
+	select LWM2M_APP_PRESS_SENSOR
+
+config APP_HUMID_SENSOR
+	bool "Enable humidity sensor"
+	select LWM2M_IPSO_HUMIDITY_SENSOR
+	select LWM2M_APP_HUMID_SENSOR
+
+config APP_GAS_RES_SENSOR
+	bool "Enable gas resistance sensor"
+	select LWM2M_IPSO_GENERIC_SENSOR
+	select LWM2M_APP_GAS_RES_SENSOR
+endif # APP_ENV_SENSOR
+endmenu # Environment sensor
+
+
+menu "Accelerometer"
+config APP_ACCELEROMETER
+	bool "Enable accelerometer"
+	select ACCELEROMETER
+	select LWM2M_IPSO_ACCELEROMETER
+	select LWM2M_APP_ACCELEROMETER
+
+if APP_ACCELEROMETER
+	rsource "src/sensors/Kconfig.accelerometer"
+endif # APP_ACCELEROMETER
+endmenu # Accelerometer
+
+menu "GPS"
+config APP_GPS
+	bool "Enable GPS"
+	select GPS
+	select LWM2M_LOCATION_OBJ_SUPPORT
+
+if APP_GPS
+config APP_GPS_HOLD_TIME
+	int "Minimum time [s] between location data notification messages"
+	default 60
+	help
+	  The minimum time in seconds that notifications can be sent to
+	  observers of the Location data.
+
+	rsource "src/sensors/Kconfig.gps"
+endif # APP_GPS
+endmenu # GPS
+endmenu # LwM2M objects
 
 config APP_LWM2M_SERVER
 	string "LwM2M server name or IP address"
 	help
 	  Choose the LwM2M server you wish to connect to.
 
-config APP_LWM2M_PSK
```

Comment:
Some of this stuff was purposely moved from C-files to Kconfig. I don't personally see a reason why it would be moved back.

Reply:
They weren't really moved back. We simply started development on a version that is older than those changes. And because we worked on it as if it were to become a new sample, these changes were never integrated. I can rewrite this so that they will be Kconfig settings though

Reply:
Ok, let's get back to this once you have done the rebase.

Reply:
I think a rebase is impossible, sadly. We were told to develop as if we were making a new sample and created the pull request with that in mind, so all previous commits are already applied. I guess you could say that the rebase happened before we applied our changes. This means that all of your changes will have to be reapplied manually. This is fine with us, but we might need some help identifying the changes that you want to keep.

Reply:
Moved stuff back to Kconfig